src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -998,6 +998,254 @@ public with sharing class ADDR_Addresses_TEST {
+        createHHTestData(hhSize, contactSize);
+
+        Contact con = listConT[0];
+        con.put('MailingCountryCode', 'US');
+        con.is_Address_Override__c = true;
+
+        Address__c addressWithoutCountry = listAddrT[0];
+        System.assertEquals('United States', listAddrT[0].MailingCountry__c);
+
+        // Set the Mailing Country to null
+        addressWithoutCountry.MailingCountry__c = null;
+        update addressWithoutCountry;
+        System.assertEquals(null, addressWithoutCountry.MailingCountry__c);
+
+        Test.startTest();
+        ADDR_Addresses_TDTM.updateEmptyAddressMailingCountryWithAccountOrContactCountryCode(con, 'Mailing', addressWithoutCountry);
+        Test.stopTest();
+
+        System.assertEquals('United States', addressWithoutCountry.MailingCountry__c);
+    }
+
+
+    /*********************************************************************************************************
+    @description
+        Update Address' Mailing State (with State & Country Picklists enabled) when the State is empty
+        and/or mismatched.
+    verify:
+        New Address' State field is updated with the source sObject's StateCode value
+    **********************************************************************************************************/
+    static testMethod void testUpdateEmptyOrMismatchedAddressMailingStateWithAccountOrContactStateCode() {
+        if (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testUpdateEmptyOrMismatchedAddressMailingStateWithAccountOrContactStateCode') {
+            return;
+        }
+
+        Integer hhSize = 1;
+        Integer contactSize = 1;
+
+        // This creates a default Address for each HH
+        createHHTestData(hhSize, contactSize);
+
+        Contact con = listConT[0];
+        con.put('MailingStateCode', 'CA');
+        con.is_Address_Override__c = true;
+
+        Address__c addressWithoutState = listAddrT[0];
+        System.assertEquals('Washington', listAddrT[0].MailingState__c);
+
+        // Set the Mailing State to null
+        addressWithoutState.MailingState__c = null;
+        update addressWithoutState;
+        System.assertEquals(null, addressWithoutState.MailingState__c);
+
+        Test.startTest();
+        ADDR_Addresses_TDTM.updateEmptyOrMismatchedAddressMailingStateWithAccountOrContactStateCode(con, 'Mailing', addressWithoutState);
+        Test.stopTest();
+
+        System.assertEquals('CA', addressWithoutState.MailingState__c);
+    }
+
+
+    /*********************************************************************************************************
+    @description
+        Update Address' Mailing State (with State & Country Picklists enabled) when the State is empty
+        and/or mismatched.
+    verify:
+        New Address' State field is updated with the source sObject's StateCode value
```

Comment:
Description and verification might need updating. The `IsMatchingStateCode`:
- Returns false if the Address MailingState is empty
- Returns false if the Address MailingState does not match to StateCode
- Returns true if the Address MailingState matches to StateCode

Test data does not have to be created in DB for this test (faster test).